quic/s2n-quic-transport/src/recovery/manager.rs

```diff
@@ -292,6 +296,16 @@ impl<Config: endpoint::Config> Manager<Config> {
     ) {
         self.pto_update_pending = false;
 
+        if self.loss_timer.is_armed() {
+            //= https://www.rfc-editor.org/rfc/rfc9002#section-6.2.1
+            //# The PTO timer MUST NOT be set if a timer is set for time threshold
+            //# loss detection; see Section 6.1.2.  A timer that is set for time
+            //# threshold loss detection will expire earlier than the PTO timer in
+            //# most cases and is less likely to spuriously retransmit data.
+            self.pto.cancel();
+            return;
```

Comment:
Ideally we'd be checking consistency in any of these return points. Maybe move the whole thing into a closure?

```rust
(|| {
  // do logic here
})();
self.check_consistency();
```

especially where we're cancelling the timer it's good to have a sanity check that the invariants still hold